### DISCUSSION TITLE
Reporting of tests conducted using textlint

### DISCUSSION THREAD

**Discussion by amimas** [textlint/textlint#1056] [Q&A]
**What version of textlint are you using?**

11.2.1

**What file type (Markdown, plain text, etc.) are you using?**
Markdown

**What did you do? Please include the actual source code causing the issue.**
Followed instructions for adding textlint and rules. There's no issue. This is an inquiry/Feature Request

I have enabled several rules enabled. It would be really nice if textlint would generate an overall report from all the rules. That would give a nice summary of the number of issues found/passed.

In Java projects, junit tests generate a report which can also produce an html report with a overall summary on the first page and then clicking on classes/methods would show more details.

Not sure if something like this already exists for textlint or if it can be easily setup using another tool. I didn't see it mentioned in the docs.

**Comment by azu** [comment]
Currently, textlint has not built-in html formatter.
But, textlint support external formatter as plugin.

- <https://textlint.github.io/docs/formatter.html>

Example: textlint formatter for lcov.

- [azu/textlint-formatter-lcov: textlint formatter for lcov format](https://github.com/azu/textlint-formatter-lcov)

We can create a html formatter like ESLint.
(textlint formatter format is very similar with ESLint's formatter)

- https://github.com/eslint/eslint/blob/master/lib/formatters/html.js

However, I have no idea for nice html reporter.(design and features)

**Comment by amimas** [comment]
Actually your references are pretty good too. I didn't even realize that there's existing reporting capabilities of overall summary of all the lints. I had probably skipped this section of the doc because it's under "formatter". Didn't expect reporting instructions to be there.

I noticed there's an option for codecov. This is mainly what I need right now. I'm going to try it out soon. Thanks for the pointers.

Html report would be nice but glad to find out there are options already available for other formats.

**Comment by amimas** [comment]
I spent a bit of time on this and tried out the `--format` option when running textlint. I like some of the formats. Definitely opens up options for me. I think by default the format is `stylish-pretty`.

I tried to see how the codecov would work but then realized I will need to link/sync my repos with codecov.io. Since I'm dealing with private repositories, I can't use this yet. So at this point I'm limited on getting coverage report.

Are there any other coverage reporting option available that I can use without relying on uploading my reports to a 3rd party service?
